src/containers/ContextProvider/reducer.js

```diff
@@ -0,0 +1,39 @@
+import { headerDefinitions } from '../../util/data';
+export const stateActions = Object.freeze({
+  updatePatient: 'update_patient',
+  updateSetting: 'update_setting', // {type, settingId, value}
+  flagStartup: 'flag_startup'
+});
+// todo: add an enum that defines possible settings
+export const reducer = (state, action) => {
+  switch (action.type) {
+    case stateActions.updateSetting:
+      return {
+        ...state,
+        [action.settingId]: action.value
+      };
+    case stateActions.updatePatient:
+      return {
+        ...state,
+        patient: action.value
+      };
+    case stateActions.flagStartup:
+      return {
+        ...state,
+        startup: true
+      };
+    default:
+      return state;
+  }
+};
+
+const initialState = {
+  patient: null,
+  startup: false,
+  redirect: ''
+};
+Object.keys(headerDefinitions).forEach(e => {
+  initialState[e] = headerDefinitions[e].default; // fill default settings values
+});
```

Comment:
Since the initialState already goes through the default field in the headerDefinitions, does that make  this code from RequestBuilder.js's componentDidMount
```js
    // init settings
    Object.keys(headerDefinitions).map(key => {
      this.setState({ [key]: headerDefinitions[key].default });
    });
```
redundant?

Reply:
they're different states but the local state isn't used so it's not redundant but it is useless. I've removed it.

Reply:
oh I missed that- I mixed up the globalDispatch for the this.setState and the other baked in state setting method